src/components/src/main/java/org/apache/jmeter/assertions/CompareAssertion.java

```diff
@@ -65,78 +65,82 @@ public AssertionResult getResult(SampleResult response) {
+                prevResult = sResult;
+            }
+            if (success) {
                 prevResult = sResult;
                 prevTime = currentTime;
+            } else {
+                result.setFailure(true);
+                buf.setLength(0);
+                appendResultDetails(buf, prevResult);
+                buf.append(JMeterUtils.getResString("comparison_response_time")).append(prevTime); //$NON-NLS-1$
+                result.addToBaseResult(buf.toString());
+                buf.setLength(0);
+                appendResultDetails(buf, sResult);
+                buf.append(JMeterUtils.getResString("comparison_response_time")).append(currentTime); //$NON-NLS-1$
+                result.addToSecondaryResult(buf.toString());
+                result.setFailureMessage(
+                        JMeterUtils.getResString("comparison_differ_time") + //$NON-NLS-1$
+                                compareTime +
+                                JMeterUtils.getResString("comparison_unit")); //$NON-NLS-1$
+                break;
             }
         }
     }
 
     private void compareContent(CompareAssertionResult result) {
-        if (compareContent) {
-            String prevContent = null;
-            SampleResult prevResult = null;
-            boolean success = true;
-            StringBuilder buf = new StringBuilder();
-            for (SampleResult sResult : responses) {
-                String currentContent = sResult.getResponseDataAsString();
-                currentContent = filterString(currentContent);
-                if (prevContent != null) {
-                    success = prevContent.equals(currentContent);
-                }
-                if (!success) {
-                    result.setFailure(true);
-                    buf.setLength(0);
-                    appendResultDetails(buf, prevResult);
-                    buf.append(prevContent);
-                    result.addToBaseResult(buf.toString());
-                    buf.setLength(0);
-                    appendResultDetails(buf, sResult);
-                    buf.append(currentContent);
-                    result.addToSecondaryResult(buf.toString());
-                    result.setFailureMessage(JMeterUtils.getResString("comparison_differ_content")); //$NON-NLS-1$
-                    break;
-                }
+        if (!compareContent) {
+            return;
+        }
+        String prevContent = null;
+        SampleResult prevResult = null;
+        boolean success = true;
+        StringBuilder buf = new StringBuilder();
+        for (SampleResult sResult : responses) {
+            String currentContent = sResult.getResponseDataAsString();
+            currentContent = filterString(currentContent);
+            if (prevContent != null) {
+                success = prevContent.equals(currentContent);
+            }
+            if (success) {
                 prevResult = sResult;
                 prevContent = currentContent;
+            } else {
```

Comment:
Same thing: I would prefer to move short if branches to the top and avoid `else` branches at the end of the code block